crates/flox/src/commands/channel.rs

```diff
@@ -144,50 +146,136 @@ fn construct_search_params(search_term: &str, flox: &Flox) -> Result<SearchParam
+    package: String,
+    /// The package description
+    description: Option<String>,
+    /// Whether to join the `input` and `package` fields when rendering
+    join: bool,
+}
+
+// FIXME: try 'python@>=2 <3', you get way more results than you should
+
+fn render_search_results_user_facing(mut search_results: SearchResults) -> Result<()> {
+    // Nothing to display
+    if search_results.results.is_empty() {
+        return Ok(());
+    }
+    // Search results contain a lot of information, but all we need for rendering are
+    // the input, the package subpath (e.g. "python310Packages.flask"), and the description.
+    let display_items = search_results
         .results
-        .iter()
+        .drain(..)
         .map(|r| {
-            let path_components = r.attr_path.clone();
-            let flake_attr = if !DEFAULT_CHANNELS.contains_key(r.input.as_str()) {
-                vec![r.input.clone(), path_components.join(".")].join("#")
-            } else {
-                path_components.join(".")
-            };
-            let d = if let Some(ref d) = r.description {
-                // Some package descriptions contain newline characters
-                // which breaks formatting
-                d.replace('\n', " ")
-            } else {
-                "<no description provided>".into()
-            };
-            (flake_attr, d)
+            // let join = !DEFAULT_CHANNELS.contains_key(r.input.as_str());
+            let join = false;
+            Ok(DisplayItem {
+                input: r.input,
+                package: r.pkg_subpath.join("."),
+                description: r.description.map(|s| s.replace('\n', " ")),
+                join,
+            })
         })
-        .collect::<Vec<_>>();
-    let attr_col_width = summarized_results
+        .collect::<Result<Vec<_>>>()?;
+
+    let deduped_display_items = dedup_and_disambiguate_display_items(display_items);
+    if deduped_display_items.is_empty() {
+        bail!("deduplicating search results failed");
+    }
+
+    let column_width = deduped_display_items
         .iter()
-        .fold(0, |cw, (attr, _)| usize::max(cw, attr.len()));
-    // Depending on the search query there could be a ton of results, better to
-    // do buffered writes than to lock `stdout` on every write.
+        .map(|d| {
+            if d.join {
+                vec![d.input.as_str(), d.package.as_str()]
+                    .join(SEPARATOR)
+                    .len()
+            } else {
+                d.package.len()
+            }
+        })
+        .max()
+        .unwrap(); // SAFETY: could panic if `deduped_display_items` is empty, but we know it's not
+
+    // Finally print something
     let mut writer = BufWriter::new(std::io::stdout());
-    for (attr, desc) in summarized_results.iter() {
-        writeln!(&mut writer, "{attr:<attr_col_width$}  {desc}")?;
+    let default_desc = String::from("<no description provided>");
```

Comment:
Does this print `my-pkgs:foo  <no description provided>` for the user to see?
Or is that just placeholder?

Reply:
Yes, for lack of any real guidance on what the output should be for missing descriptions. It would be trivial to change it to not displaying anything.

Reply:
I'd recommend leaving it blank which is existing behavior, but I don't have strong feelings about it.